lib/routes/cool18/index.ts

```diff
@@ -3,73 +3,111 @@ import cache from '@/utils/cache';
 import got from '@/utils/got';
 import { load } from 'cheerio';
 import { parseDate } from '@/utils/parse-date';
+import type { Context } from 'hono';
+import ofetch from '@/utils/ofetch';
+
+type PageDataItem = {
+    tid: string;
+    username: string;
+    subject: string;
+    dateline: string;
+    type: string;
+};
 
 export const route: Route = {
-    path: '/:id?/:type?/:keyword?',
+    path: '/:id/:type?/:keyword?/:pageSize?',
```

Comment:
Why change `id` to compulsory parameter? This forces current users who subscribe `/cool18` need to update their RSS reader.